design/_template.md

```diff
@@ -0,0 +1,147 @@
+
+Introduction paragraph -- why are we doing anything? A single
+paragraph of prose that users can understand. The title and this first
+paragraph should be used as the subject line and body of the commit
+message respectively.
+
+Some notes about the design process:
+
+* Not all changes need a design document. Anything that is likely to
+  result in significant review discussion, or changes to user-facing
+  documentation, does.
+
+* The aim of this document is first to define the problem we need to
+  solve, and second agree the overall approach to solve that problem.
+
+* This document is not intended to be extensive user-facing
+  documentation for a new feature. It should provide enough detail for
+  contributors to the project to understand the impact and help refine
+  the design.
+
+Some notes about using this template:
+
+* Your spec should be in [Markdown
+  syntax](https://daringfireball.net/projects/markdown/syntax), like
+  this template.
+
+* Please wrap text at 79 columns.
+
+* Please do not delete any of the sections in this template.  If you have
+  nothing to say for a whole section, just write: None
+
+
+# Problem description #
+
+A detailed description of the problem. What problem is this blueprint
+addressing?
+
+## Use Cases ##
+
+What use cases does this address? What impact on actors does this
+change have?  Ensure you are clear about the actors in each use case:
+Developer, End User, Deployer etc.
+
+# Proposed change #
+
+Here is where you cover the change you propose to make in detail. How
+do you propose to solve this problem?
+
+If this is one part of a larger effort make it clear where this piece
+ends. In other words, what's the scope of this effort?
+
+## Alternatives ##
+
+What other ways could we solve the problem? Why should we not use
+those approaches? This doesn't have to be a full literature review,
+but it should demonstrate that thought has been put into why the
+proposed solution is an appropriate one.
+
+## CRD impact ##
+
+* What new objects is this going to require?
+* What new fields are needed, for either the Spec or Status section?
+* What other changes (labels, annotations, etc.) are needed?
+
+## Security impact ##
+
+Describe any potential security impact on the system.
+
+* Does this change touch sensitive data such as credentials or other
+  secret values?
+
+* Does this change involve cryptography or hashing?
+
+* Does this change require the use of any elevated privileges?
+
+## Other end user impact ##
+
+Aside from the API, are there other ways a user will interact with this
+feature?
+
+## Performance Impact ##
+
+Describe any potential performance impact on the system, for example
+how often will new code be called, and is there a major change to the calling
+pattern of existing code.
+
+# Implementation #
+
+## Assignee(s) ##
+
+Who is leading the writing of the code?
+
+## Work Items ##
+
+Work items or tasks -- break the feature up into the things that need to be
```

Comment:
Would it make sense to link the assignees to the work items?

Reply:
Probably. They might not all be known to start, but that information could certainly be included as people pick up the tasks. We could also use github issues for tracking that, and I would expect us to link to those issues in that case.